### PR TITLE
[OSX] fix make deploy when compiling on OSX

### LINKED ISSUE #8120: [OSX] `make deploy` is currently failing
OS X 10.11.5, Python 3.5.1
After building as usual.

```
$ make deploy
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C src qt/bitcoin-qt
/usr/local/bin/python3 ./contrib/macdeploy/macdeployqtplus Bitcoin-Qt.app -add-qt-tr da,de,es,hu,ru,uk,zh_CN,zh_TW -translations-dir= -dmg -fancy ./contrib/macdeploy/fancy.plist -verbose 2 -volname Bitcoin-Core
Error: Could not find background picture at "background.tiff" or "./contrib/macdeploy/background.tiff"
make: *** [Bitcoin-Core.dmg] Error 1
```

The instructions in [macdeploy/README.md](https://github.com/bitcoin/bitcoin/blob/master/contrib/macdeploy/README.md) might also need updating.

### PR DESCRIPTION
**Pull request by jonasschnelli**
Should fix #8120.
There is a ugly issue of a python sub-package incompatibility https://github.com/bitcoin/bitcoin/issues/8134

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -68,6 +68,7 @@ AC_PATH_TOOL(GCOV, gcov)
 AC_PATH_PROG(LCOV, lcov)
 AC_PATH_PROG(JAVA, java)
 AC_PATH_PROGS([PYTHON], [python3 python2.7 python2 python])
+AC_PATH_PROGS([PYTHON2], [python2.7 python2 python])
```
Comment (jonasschnelli) [inline configure.ac]: This is not really elegant. Fixing the upstream pip package would probably the way to go.

**Comment by fanquake** [comment]
This seems to work, but still required some playing around with pip to get the correct packages installed. It didn't seem to find the brew installed python2 packages?

Could you rebase the doc changes off #8029 ?
Also, should we just delete the readme in contrib/macdeploy ?

**Comment by jonasschnelli** [comment]
Rebased on now merged #8029, simplified contrib/macdeploy's readme,

File: configure.ac (hunk 1 above)
Comment (laanwj) [inline configure.ac]: If this requires everyone to have Python 2 installed, including those that aren't building on mac, that's not really acceptable

Comment (jonasschnelli) [inline configure.ac]: Yes. Totally agree, ... I'm testing now the solution from @theuni (https://github.com/bitcoin/bitcoin/issues/8120#issuecomment-223691744).

Comment (theuni) [inline configure.ac]: At this point, I'm thinking the easiest thing to do is just pull mac_alias and ds_store into our repo in macdeploy. They're small/simple enough, and that saves us a ton of complexity.

**Comment by jonasschnelli** [comment]
ping @theuni, does this still make sense? I guess https://github.com/bitcoin/bitcoin/issues/8120 is still unfixed?

**Comment by droark** [comment]
Ahhh, I just stumbled across this issue today. Glad to see that people are looking into it.

Having looked at the proposed solutions, I think theuni's proposal that ds_store and mac_alias be pulled into the repo is reasonable. They're ~1300 lines of Python code each. Not tiny but quite manageable.

**Comment by jonasschnelli** [comment]
Closing in the hope @theuni will PR https://github.com/theuni/bitcoin/commit/c622254170d4bd8c5ca75077214105cef666f4c4.

**Pull request closed without merging**